atom/browser/atom_cert_verifier.h

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2015 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ATOM_BROWSER_ATOM_CERT_VERIFIER_H_
+#define ATOM_BROWSER_ATOM_CERT_VERIFIER_H_
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "base/memory/ref_counted.h"
+#include "net/base/hash_value.h"
+#include "net/cert/cert_verifier.h"
+#include "net/cert/crl_set.h"
+#include "net/cert/x509_certificate.h"
+#include "net/log/net_log.h"
+
+namespace atom {
+
+class AtomCertVerifier : public net::CertVerifier {
+ public:
+  struct RequestParams {
+    RequestParams(
+        const net::SHA1HashValue cert_fingerprint,
+        const net::SHA1HashValue ca_fingerprint,
+        const std::string& hostname_arg,
+        const std::string& ocsp_response,
+        int flags);
+    ~RequestParams() {}
+
+    bool operator<(const RequestParams& other) const;
+
+    std::string hostname;
+    std::string ocsp_response;
+    int flags;
+    std::vector<net::SHA1HashValue> hash_values;
+  };
+
+  class CertVerifyRequest
+      : public net::CertVerifier::Request,
```

Comment:
On async completion `out_req` need to be set to the request in process otherwise it will be cancelled https://code.google.com/p/chromium/codesearch#chromium/src/net/cert/cert_verifier.h&q=cert_verifier.h&sq=package:chromium&l=108 . Totally misunderstood the use case, they use it to manage the lifetime of requests create but in our case i was already managing using a reference to it in a set. `out_req` should not be needed here and the inheritence was just for that, can remove it. The `Release` call was used to mark the end of lifetime of a single `CertVerifyRequest` , is it a bad practice ?